File: pronacej-front/src/app/modules/seguridad/services/informeTecnicoSustentatorio.service.ts
import { Injectable } from '@angular/core';
import { InformeTecnicoSustentatorioDTO } from 'app/core/model/both/informeTecnicoSustentatorioDTO.model';
import { PaginacionRequest } from 'app/core/model/request/PaginacionRequest.model';
import { PaginacionResponse } from 'app/core/model/response/PaginacionResponse.model';
import { RespuestaPorDefecto } from 'app/core/model/response/RespuestaPorDefecto.model';
import { BackendService } from 'app/core/services/backend.service';
import { Observable } from 'rxjs';

@Injectable({
    providedIn: 'root',
})
export class InformeTecnicoSustentatorioService {
    private path = '/informe-tecnico';

    constructor(private backendService: BackendService) {}

    /**
     * Obtiene los informes técnicos disponibles para el sistema
     *
     * @param paginacionRequest Datos de paginación
     * @param nemonicoMenu string nemonico de un informe técnico del sistema
     *
     * @returns Observable<RespuestaPorDefecto<PaginacionResponse<InformeTecnicoSustentatorioDTO>>>
     */
    obtenerInformesTecnicosPaginado(
        paginacionRequest: PaginacionRequest,
        nemonicoMenu = ''
    ): Observable<
        RespuestaPorDefecto<PaginacionResponse<InformeTecnicoSustentatorioDTO>>
    > {
        let endPoint = this.path + '/obtenerInformesTecnicosPaginado';
        return this.backendService.postFinal(
            endPoint,
            paginacionRequest,
            nemonicoMenu
        );
    }

    /**
     * Crea un informe técnico en el sistema con los datos enviados en el request
     *
     * @param informeTecnicoDTO datos del informe técnico a crear
     * @param nemonicoMenu string nemonico de un menu del sistema
     *
     * @returns Observable<RespuestaPorDefecto<InformeTecnicoSustentatorioDTO>>
     */
    crearInformeTecnico(
        informeTecnicoDTO: InformeTecnicoSustentatorioDTO,
        nemonicoMenu: string = ''
    ): Observable<RespuestaPorDefecto<InformeTecnicoSustentatorioDTO>> {
        let endPoint = this.path + '/crearInformeTecnico';
        return this.backendService.postFinal(
            endPoint,
            informeTecnicoDTO,
            nemonicoMenu
        );
    }

    /**
     * Elimina un informe técnico en el sistema con los datos enviados en el request
     *
     * @param informeTecnicoDTO datos del informe técnico a eliminar
     * @param nemonicoMenu string nemonico de un menu del sistema
     *
     * @returns Observable<RespuestaPorDefecto<boolean>>
     */
    eliminarInformeTecnico(
        informeTecnicoDTO: InformeTecnicoSustentatorioDTO,
        nemonicoMenu = ''
    ): Observable<RespuestaPorDefecto<boolean>> {
        let endPoint = this.path + '/eliminarInformeTecnico';
        return this.backendService.postFinal(
            endPoint,
            informeTecnicoDTO,
            nemonicoMenu
        );
    }

    async checkError(error: any, mostrarError = true): Promise<string> {
        return await this.backendService.checkError(error, mostrarError);
    }
}

File: pronacej-backend-seguridad/src/main/java/net/latinus/sistema/integral/gestion/seguridad/service/IA/InformeTecnicoSustentatorioService.java
package net.latinus.sistema.integral.gestion.seguridad.service.IA;

import jakarta.servlet.http.HttpServletRequest;
import net.latinus.sistema.integral.gestion.seguridad.model.both.BodyEncriptado;
import net.latinus.sistema.integral.gestion.seguridad.model.both.InformeTecnicoSustentatorioDTO;
import net.latinus.sistema.integral.gestion.seguridad.model.response.PaginacionResponse;
import net.latinus.sistema.integral.gestion.seguridad.model.response.RespuestaPorDefectoAuditoria;

public interface InformeTecnicoSustentatorioService {
    
    /**
     * Devuelve un objeto RespuestaPorDefectoAuditoria con data InformeTecnicoSustentatorioDTO Paginada
     *
     * @param httpServletRequest request peticion.
     * @param bodyEncriptado PaginacionRequest datos para obtener todos los informes técnicos.
     *
     * @return RespuestaPorDefectoAuditoria<InformeTecnicoSustentatorioDTO>
     */
    RespuestaPorDefectoAuditoria<PaginacionResponse<InformeTecnicoSustentatorioDTO>> obtenerInformesTecnicosPaginado(
            HttpServletRequest httpServletRequest,
            BodyEncriptado bodyEncriptado);
    
    /**
     * Devuelve un objeto RespuestaPorDefectoAuditoria con data InformeTecnicoSustentatorioDTO si el informe técnico se creó con éxito
     *
     * @param httpServletRequest request petición.
     * @param bodyEncriptado objeto InformeTecnicoSustentatorioDTO a crear.
     *
     * @return RespuestaPorDefectoAuditoria<InformeTecnicoSustentatorioDTO>
     */
    RespuestaPorDefectoAuditoria<InformeTecnicoSustentatorioDTO> crearInformeTecnico(
            HttpServletRequest httpServletRequest,
            BodyEncriptado bodyEncriptado);
    
    /**
     * Elimina un informe técnico del sistema
     *
     * @param httpServletRequest HttpServletRequest datos del request.
     * @param bodyEncriptado PaginacionRequest datos del informe técnico a eliminar.
     *
     * @return RespuestaPorDefectoAuditoria<Boolean>
     */
    RespuestaPorDefectoAuditoria<Boolean> eliminarInformeTecnico(
            HttpServletRequest httpServletRequest, 
            BodyEncriptado bodyEncriptado);

}